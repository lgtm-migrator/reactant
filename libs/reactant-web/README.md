# reactant-web

![Node CI](https://github.com/unadlib/reactant/workflows/Node%20CI/badge.svg)

A Web application with react-dom for Reactant

## Usage

```sh
npm install reactant-web
yarn add reactant-web
```

You can visit [reactant.js.org](https://reactant.js.org/) for more documentation.
