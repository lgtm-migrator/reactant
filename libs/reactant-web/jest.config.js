module.exports = {
  displayName: 'reactant-web',
  preset: '../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': 'babel-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../coverage/libs/reactant-web',
};
