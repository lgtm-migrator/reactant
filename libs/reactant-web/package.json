{
  "name": "reactant-web",
  "version": "0.14.0",
  "description": "A Web application with react-dom for Reactant",
  "author": "unadlib",
  "license": "MIT",
  "homepage": "https://github.com/unadlib/reactant#readme",
  "repository": {
    "type": "git",
    "url": "https://github.com/unadlib/reactant.git",
    "directory": "packages/reactant-storage"
  },
  "bugs": {
    "url": "https://github.com/unadlib/reactant/issues"
  },
  "dependencies": {
    "reactant-router-dom": "^0.14.0"
  }
}
