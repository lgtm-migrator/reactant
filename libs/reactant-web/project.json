{
  "root": "libs/reactant-web",
  "sourceRoot": "libs/reactant-web/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "build": {
      "executor": "@nrwl/web:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/reactant-web",
        "tsConfig": "libs/reactant-web/tsconfig.lib.json",
        "project": "libs/reactant-web/package.json",
        "entryFile": "libs/reactant-web/src/index.ts",
        "external": ["react/jsx-runtime"],
        "updateBuildableProjectDepsInPackageJson": false,
        "rollupConfig": "@nrwl/react/plugins/bundle-rollup",
        "compiler": "babel",
        "assets": [
          {
            "glob": "libs/reactant-web/README.md",
            "input": ".",
            "output": "."
          }
        ]
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/reactant-web/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/reactant-web"],
      "options": {
        "jestConfig": "libs/reactant-web/jest.config.js",
        "passWithNoTests": true
      }
    }
  }
}
