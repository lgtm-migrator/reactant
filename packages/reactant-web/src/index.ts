/* eslint-disable camelcase */
import * as ReactDom from 'react-dom';

export type { Renderer } from 'react-dom';

export {
  BrowserRouter,
  MemoryRouter,
  Switch,
  Route,
  Link,
  useRouteMatch,
  useParams,
  useHistory,
  useLocation,
  generatePath,
  Prompt,
  Redirect,
  Router,
  StaticRouter,
  matchPath,
  withRouter,
} from 'reactant-router-dom';

export type {
  RedirectProps,
  RouteChildrenProps,
  RouteComponentProps,
  SwitchProps,
  match,
  RouterChildContext,
  RouteProps,
} from 'reactant-router-dom';

export const {
  findDOMNode,
  unmountComponentAtNode,
  createPortal,
  version,
  render,
  hydrate,
  unstable_batchedUpdates,
  unstable_renderSubtreeIntoContainer,
} = ReactDom;
