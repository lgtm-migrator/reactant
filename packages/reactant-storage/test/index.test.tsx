/* eslint-disable no-shadow */
/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { act } from 'react-dom/test-utils';

import {
  ViewModule,
  createApp,
  injectable,
  inject,
  optional,
  useConnector,
  action,
  computed,
  autobind,
  state,
} from 'reactant';
import { render, unmountComponentAtNode } from 'reactant-web';
import { Storage, StorageOptions, IStorageOptions } from '..';

let container: Element;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
});

describe('base API', () => {
  test('base persistence module', async () => {
    @injectable()
    class Bar {
      constructor(public storage: Storage) {
        this.storage.setStorage(this, {
          whitelist: ['test'],
        });
      }

      name = 'bar';

      @state
      test = 'test';
    }

    @injectable()
    class Count {
      constructor(public storage: Storage) {
        this.storage.setStorage(this, {
          blacklist: ['num'],
        });
      }

      name = 'count';

      @state
      num = 0;

      @state
      num1 = 0;

      @autobind
      @action
      increase() {
        this.num += 1;
      }
    }

    @injectable()
    class DashboardView extends ViewModule {
      constructor(public count: Count) {
        super();
      }

      @computed((that: DashboardView) => [that.count.num])
      get num() {
        return this.count.num + 1;
      }

      getData = () => ({
        num: this.num,
        increase: this.count.increase,
      });

      component() {
        const data = useConnector(this.getData);
        return (
          <div onClick={data.increase} id="increase">
            {data.num}
          </div>
        );
      }
    }

    @injectable()
    class AppView extends ViewModule {
      constructor(public bar: Bar, public dashboardView: DashboardView) {
        super();
      }

      component() {
        return <this.dashboardView.component />;
      }
    }
    const storage = {
      data: {} as Record<string, any>,
      getItem(key: string): Promise<string> {
        return new Promise((resolve) => {
          resolve(this.data[key]);
        });
      },
      setItem(key: string, item: string): Promise<void> {
        return new Promise((resolve) => {
          this.data[key] = item;
          resolve();
        });
      },
      removeItem(key: string): Promise<void> {
        return new Promise((resolve) => {
          delete this.data[key];
          resolve();
        });
      },
    };
    const app = createApp({
      modules: [
        {
          provide: StorageOptions,
          useValue: {
            storage,
          } as IStorageOptions,
        },
      ],
      main: AppView,
      render,
    });
    act(() => {
      app.bootstrap(container);
    });
    await new Promise((resolve) => {
      setTimeout(resolve);
    });
    expect(container.querySelector('#increase')?.textContent).toBe('1');
    act(() => {
      container
        .querySelector('#increase')!
        .dispatchEvent(new MouseEvent('click', { bubbles: true }));
    });
    expect(container.querySelector('#increase')?.textContent).toBe('2');
    await new Promise((resolve) => {
      setTimeout(resolve, 100);
    });
    expect(app.store?.getState()).toEqual({
      bar: { test: 'test', _persist: { version: -1, rehydrated: true } },
      count: { num: 1, num1: 0, _persist: { version: -1, rehydrated: true } },
      _persist: { version: -1, rehydrated: true },
    });
    expect(storage.data).toEqual({
      'persist:root': '{"_persist":"{\\"version\\":-1,\\"rehydrated\\":true}"}',
      'persist:bar':
        '{"test":"\\"test\\"","_persist":"{\\"version\\":-1,\\"rehydrated\\":true}"}',
      'persist:count':
        '{"num1":"0","_persist":"{\\"version\\":-1,\\"rehydrated\\":true}"}',
    });
  });
});
