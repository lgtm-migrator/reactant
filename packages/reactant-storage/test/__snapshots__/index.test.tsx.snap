// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base API base persistence module with non-string name 1`] = `"Module 'AppView' is invalid for using 'setStorage', the parameter 'options.name' of the decorator '@injectable(options)' that decorates the 'AppView' module must be specified as a string."`;

exports[`base API base persistence module without any state 1`] = `"Module 'AppView' is invalid for using 'setStorage', the current module does not have any global state that is decorated with '@state'."`;

exports[`base API base persistence module without setting name 1`] = `"Module 'AppView' is invalid for using 'setStorage', the parameter 'options.name' of the decorator '@injectable(options)' that decorates the 'AppView' module must be specified as a string."`;

exports[`check with error base persistence module with \`_persist\` state 1`] = `"Module 'AppView' is invalid for using 'setStorage', the current module should not customize the state with the key '_persist'."`;

exports[`check with error base persistence with \`_persist\` module 1`] = `"Storage module conflict with the names of other modules, make sure that no module has a name of '_persist'."`;
